File: scripts/N.pl
#!/usr/bin/perl
use strict;
die "Usage: perl $0 <fasta_file> <out_file>" if (@ARGV!=2);
$a=shift;
$b=shift;
open A,"$a";
open O,">$b";
$/=">";
<A>;
$/="\n";
my $N=0;my $tot=0;
while (<A>){
		my $title = $_;
		my $seq_name = $1 if($title =~ /^(\S+)/);
		$/=">";
		my $seq=<A>;
		chomp $seq;
		$/="\n";
		$seq=~s/\s//g;
        $tot+=length($seq);
		my $n;
		$n=$seq=~ s/N/N/ig ;
		$N+=$n;
#		print "$seq_name\tN length:$n\n";
}

print O"total length:$tot\nN length:$N\n";

File: scripts/N.py
#!/usr/env/python
import re
import sys

if len(sys.argv)<2 :
    print("Usage: python "+sys.argv[0]+" fasta_file")
    exit()
file=open(sys.argv[1],'r')
total=0
N=0
for line in file:
    if re.search('>',line) :
        continue
    line=line.strip()
    total+=len(line)
    line=line.upper()
    N+=line.count('N')

print ('Total: '+str(total))
print ('N|n_num: '+str(N) )
proportion=N/total
print ('N|n_num / Total = %.3f' % proportion )
